gitgud/skills/rewriting/_truth/explanation.txt

```diff
@@ -1,12 +1,42 @@
-You've already seen some of the stuff "git rebase -i" can do.
-"git rebase -i" can also be used to reoder commits.
+There's something you need to know about...
 >>>
-When the editor opens, simply re-order the commits to be in the order you want them to be in.
-On a technical level, Git is changing the parent of each of the commits. The order of the commits is determined by looking at the parents of each commit.
+The truth is...
 >>>
-For this level, each commit has a single word message that are meant to come together to form the sentence "This is an easy level" when viewing them in chronological order. You are meant to put the commits in order and form that sentence.
+What you did with "git rebase -i"...
 >>>
-You can see the commits in chronological order by running "git log --reversed".
-Git normally tries to show you the most recent commit first since that's normally more useful. That's why you need the "--reversed" flag
+Can be done by...
 >>>
-Run "git rebase -i <commit>" for more details. It contains useful information
+Detatching your head! Eek!
+>>>
+Let me clarify, I'm actually talking about your HEAD.
+>>>
+HEAD is used to refer to the commit or branch that is currently checked out.
+This basically means that when you're on branch master, HEAD means master, which itself means a commit.
+>>>
+This is important because when you're in a 'detached HEAD' state, cherry-picking still works the same way, and you can do exactly what you just did in the last level.
+>>>
+Normally, HEAD points to the branch that you have checked out, so it refers to whichever commit the branch does, even if there is a new commit!
+For example, if you have bugFix checked out, then when bugFix is updated, HEAD is updated as well, just like this:
+
+    HEAD -> bugFix -> C1   // HEAD points to bugFix.
+    HEAD -> bugFix -> C2   // A commit moved bugFix to C2, HEAD effectively points to C2
```

Comment:
Do you think we could change this diagram so that we could show multiple commits? Something like
```
// HEAD points to bugFix.
C1             
^ bugFix < HEAD

// A new commit (C2) caused bugFix to move to C2, HEAD effectively points to C2
C1 ---- C2 
        ^ bugFix < HEAD
```

I think this better illustrates the point that HEAD will point to the branch, even if there's a new commit.